common/types/fixationEntry.go

```diff
@@ -1,20 +1,41 @@
 package types
 
 import (
+	"math"
+
 	sdk "github.com/cosmos/cosmos-sdk/types"
 )
 
-func FixationVersion() uint64 {
-	return 3
-}
-
 // IsStale tests whether an entry is stale, i.e. has refcount zero _and_
 // has passed its stale_at time (more than STALE_ENTRY_TIME since deletion).
-func (entry Entry) IsStale(ctx sdk.Context) bool {
+func (entry Entry) IsStaleBy(block uint64) bool {
 	if entry.GetRefcount() == 0 {
-		if entry.StaleAt <= uint64(ctx.BlockHeight()) {
+		if entry.StaleAt <= block {
 			return true
 		}
 	}
 	return false
 }
+
+// IsDeletedBy tests whether an entry is deleted, with respect to a given
+// block, i.e. has entry.DeletAt smaller or equal to that that block.
+func (entry Entry) IsDeletedBy(block uint64) bool {
+	return entry.DeleteAt <= block
+}
+
+// IsDeleted tests whether an entry is currently deleted, i.e. has its
+// entry.DeleteAt below the current ctx block height.
+func (entry Entry) IsDeleted(ctx sdk.Context) bool {
+	return entry.IsDeletedBy(uint64(ctx.BlockHeight()))
+}
+
+// IsMarkedDeleted tests whether an entry is marked for deletion, i.e. has
+// entry.Deleted != math.MaxUint64.
+func (entry Entry) HasDeleteAt() bool {
+	return entry.DeleteAt < math.MaxUint64
+}
+
+// IsEntryIndexLive tests whether an entry-index is live
+func IsEntryIndexLive(value []byte) bool {
+	return value[0] == EntryIndexLive[0]
```

Comment:
index 0? const

Reply:
the value here is 1 byte.  but the KVstore deals with []byte, so the code uses the first one. always.
by the way, for existing entries - that is valid ascii, so cannot be 0, or 1, that's why we don't need migration. over time these entries will disappear...